### PR TITLE
bump: bump up oras-go and image-spec

### PR DESCRIPTION
**Pull request by Two-Hearts**
bumping up oras-go to v2.4.0 and image-spec to v1.1.0-rc6.

Removed dependency of `oras-credentials-go` as it's moved to `oras-go v2.4.0`.

### CODE REVIEW

**Review by Wwwsylvia — APPROVED** [review]
LGTM, but I am not a maintainer

**Review by JeyJeyGao — DISMISSED** [review]
LGTM

File: cmd/notation/login.go [hunk 1]
```diff
@@ -25,9 +25,9 @@ import (
 	"github.com/notaryproject/notation-go/log"
 	"github.com/notaryproject/notation/internal/auth"
 	"github.com/notaryproject/notation/internal/cmd"
-	credentials "github.com/oras-project/oras-credentials-go"
 	"github.com/spf13/cobra"
 	"golang.org/x/term"
+	credentials "oras.land/oras-go/v2/registry/remote/credentials"
```
Comment (priteshbandi) [inline cmd/notation/login.go:30]: nit: we dont need to use import alias.

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by shizhMSFT — COMMENTED** [review]
LGTM

**Pull request merged**